site/src/concurrency-primitives.md

```diff
@@ -0,0 +1,292 @@
+---
+layout: page
+title:  "Concurrency Primitives"
+section: "concurrency-primitives"
+position: 2
+---
+
+# Concurrency Primitives
+
+In the [`fs2.async` package object](https://github.com/functional-streams-for-scala/fs2/blob/series/0.10/core/shared/src/main/scala/fs2/async/async.scala) you'll find a bunch of useful concurrency primitives:
+
+- `Ref[F, A]`
+- `Promise[F, A]`
+- `Queue[F, A]`
+- `Topic[F, A]`
+- `Signal[F, A]`
+- `Semaphore[F]`
+
+These data structures could be very handy in a few cases scenarios. See below examples of each of them originally taken from [gvolpe examples](https://github.com/gvolpe/advanced-http4s/tree/master/src/main/scala/com/github/gvolpe/fs2):
```

Comment:
both saying cases and scenarios is redundant :)

Reply:
Maybe it should be "case scenario" instead of "cases scenarios" but as far as I know it's normal to use them in conjunction.

Reply:
Definitely agree this a little weird as written.